contracts/Staking.sol

```diff
@@ -0,0 +1,487 @@
+     */
+    function totalStakedForAt(address _account, uint256 _blockNumber) public view returns (uint256) {
+        return stakedHistory[_account].get(_blockNumber);
+    }
+
+    /**
+     * @notice Get the total amount of tokens staked by all users at block number `_blockNumber`
+     * @param _blockNumber Block number at which we are requesting
+     * @return The amount of tokens staked at the given block number
+     */
+    function totalStakedAt(uint256 _blockNumber) public view returns (uint256) {
+        return totalStakedHistory.get(_blockNumber);
+    }
+
+    /**
+     * @notice Unlock `_account`'s lock #`_lockId` so locked tokens can be unstaked again
+     * @param _account Owner of locked tokens
+     * @param _lockId Id of the lock for the given account
+     */
+    function unlock(address _account, uint256 _lockId) public {
+        // only manager and owner (if manager allows) can unlock
+        require(canUnlock(_account, _lockId), ERROR_CAN_NOT_UNLOCK);
+
+        _unlock(_account, _lockId);
+    }
+
+    /**
+     * @notice Get the staked but not locked amount of tokens by `_account`
+     * @param _account Owner of the staked but unlocked balance
+     * @return Amount of tokens staked but not locked by given account
+     */
+    function unlockedBalanceOf(address _account) public view returns (uint256) {
+        uint256 unlockedTokens = totalStakedFor(_account);
+
+        Account storage account = accounts[_account];
+        for (uint256 i = 0; i < account.activeLockIds.length; i++) {
+            unlockedTokens = unlockedTokens.sub(account.locks[account.activeLockIds[i]].amount);
+        }
+
+        return unlockedTokens;
+    }
+
+    /**
+     * @notice Check if `_account`'s lock #`_lockId` can be unlocked
+     * @param _account Owner of lock
+     * @param _lockId Id of the lock for the given account
+     * @return Whether given lock of given account can be unlocked
+     */
+    function canUnlock(address _account, uint256 _lockId) public view returns (bool) {
+        Lock storage _lock = accounts[_account].locks[_lockId];
+
+        if (msg.sender == address(_lock.manager) ||
+            (msg.sender == _account && _lock.manager.canUnlock(_account, _lockId, _lock.data))) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /*
+    function multicall(bytes[] _calls) public {
+        for(uint i = 0; i < _calls.length; i++) {
+            require(address(this).delegatecall(_calls[i]), ERROR_MULTICALL_DELEGATECALL);
+        }
+    }
+    */
+
+    /* Internal functions */
+
+    function _stakeFor(address _account, uint256 _amount, bytes _data) internal {
+        // stake 0 tokens makes no sense
+        require(_amount > 0, ERROR_AMOUNT_ZERO);
+
+        // transfer tokens
+        require(stakingToken.transferFrom(msg.sender, this, _amount), ERROR_TOKEN_TRANSFER);
+
+        // process Stake
+        _modifyStakeBalance(_account, _amount, true);
```

Comment:
Yes, see: https://github.com/aragon/staking/pull/5/files
(sorry, I forgot to assign you as reviewer there)
I put separately because Remix was telling me that it was better with the return value, but embark tests were showing more gas consumption. I think it's because embark shows all gas consumed, including deployment, but wanted to make sure.